InteroperabilityTest_Engine/Compute/Compare.cs

```diff
@@ -0,0 +1,391 @@
+
+                    foreach (oM.Test.ITestInformation innerInformation in result.NonEventMessageInformation())
+                    {
+                        //Find matching reference data
+                        oM.Test.ITestInformation reference = referenceInformation.IGetSameIdInformation(innerInformation);
+
+                        //Compare the inner test information
+                        oM.Test.ITestInformation comparisonResult = ICompare(innerInformation, reference);
+                        if (comparisonResult != null)
+                            diffResult.Information.Add(comparisonResult);
+
+                        if (reference != null)
+                            referenceInformationNoMatch.Remove(reference);
+                    }
+
+                    foreach (oM.Test.ITestInformation refInfo in referenceInformationNoMatch)
+                    {
+                        diffResult.Information.Add(refInfo.IOnlyReferenceFound());
+                    }
+                }
+                else
+                {
+                    //Else, the results are the same
+                    diffResult.Message = "No change in the result status. Was and still is: " + result.Status + ". No changes in differences found from reference data.";
+                    diffResult.Status = oM.Test.TestStatus.Pass;
+                    return diffResult;
+                }
+            }
+            //Result Status are different
+            else
+            {
+                //Reference is pass, current is not
+                if (refResult.Status == oM.Test.TestStatus.Pass)
+                {
+                    if (result.Status == oM.Test.TestStatus.Error)
+                    {
+                        //System is now crashing, was working before
+                        diffResult.Message = "An error has been introduced that was previously fully passing!";
+                        diffResult.Status = oM.Test.TestStatus.Error;
+                        return diffResult;
+                    }
+                    else if (result.Status == oM.Test.TestStatus.Warning)
+                    {
+                        //New differences previously not present have been introduced
+                        foreach (oM.Test.ITestInformation info in result.NonEventMessageInformation())
+                        {
+                            diffResult.Information.Add(info.INoReferenceFound());
+                        }
+                        diffResult.Message = "Object previously showing up as a pass is now showing warnings!";
+                        diffResult.Message += Environment.NewLine + diffResult.ExceptionProperties(true, true, oM.Test.TestStatus.Warning);
+                        diffResult.Status = oM.Test.TestStatus.Error;
+                        return diffResult;
+                    }
+                }
+                //Reference is exception, current is not
```

Comment:
This comment should be within whichever set of brackets it applies to - I can't work out which set it goes with.